src/ripple/conditions/impl/Fulfillment.cpp

```diff
@@ -0,0 +1,229 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/conditions/Condition.h>
+#include <ripple/conditions/Fulfillment.h>
+#include <ripple/conditions/PreimageSha256.h>
+#include <ripple/conditions/PrefixSha256.h>
+#include <ripple/conditions/RsaSha256.h>
+#include <ripple/conditions/Ed25519.h>
+#include <ripple/conditions/impl/utils.h>
+#include <boost/regex.hpp>
+#include <boost/optional.hpp>
+#include <type_traits>
+#include <vector>
+
+namespace ripple {
+namespace cryptoconditions {
+
+bool
+fulfills (
+    Fulfillment const& f,
+    Condition const& c)
+{
+    // Fast check: the fulfillment's type must match the
+    // conditions's type:
+    if (f.type() != c.type)
+        return false;
+
+    // Ensure that the condition is well-formed
+    if (!validate (c))
+        return false;
+
+    // The fulfillment payload can be no larger than the
+    // what the condition allows.
+    if (f.payloadSize() > c.maxFulfillmentLength)
+        return false;
+
+    return f.condition() == c;
+}
+
+bool
+validate (
+    Fulfillment const& f,
+    Condition const& c,
+    Slice m)
+{
+    return fulfills (f, c) && f.validate (m);
+}
+
+bool
+validateTrigger (
+    Fulfillment const& f,
+    Condition const& c)
+{
+    return validate (f, c, {});
+}
+
+std::unique_ptr<Fulfillment>
+loadFulfillment (std::uint16_t type, Slice payload)
+{
+    std::unique_ptr<Fulfillment> p;
+
+    switch (type)
```

Comment:
We don't support `threshold` signatures yet - they're next on the list, and once they are supported, they'll be included here.